Lib/ufo2ft/outlineCompiler.py

```diff
@@ -978,6 +983,34 @@ def setupTable_CPAL(self):
         except ColorLibError as e:
             raise InvalidFontData("Failed to build CPAL table") from e
 
+    def setupTable_meta(self):
+        """
+        Make the meta table.
+
+        ***This should not be called externally.** Sublcasses
+        may override or supplement this method to handle the
+        table creation in a different way if desired.
+        """
+        if "meta" not in self.tables:
+            return
+
+        font = self.ufo
+        self.otf["meta"] = meta = newTable("meta")
+        ufo_meta = font.lib.get(OPENTYPE_META_KEY)
+        for key, value in ufo_meta.items():
+            if key in ["dlng", "slng"]:
+                assert isinstance(value, (list, tuple))
+                assert all(isinstance(string, str) for string in value)
+            elif key in ["appl", "bild"]:
+                assert isinstance(value, bytes)
+
+            if isinstance(value, bytes):
+                meta.data[key] = value
+            elif key in ("dlng", "slng") and isinstance(value, (list, tuple)):
+                meta.data[key] = ",".join(value)
+            elif isinstance(value, str):
+                meta.data[key] = value
```

Comment:
i suggest you do the type-checking for the registered meta tags in the same `if` statement, e.g.

```python
if key in ("dlng", "slng"):
    if not isinstance(key, (list, tuple)):
        raise TypeError(...)
    ...
elif key in ("appl", "bild"):
    if not isinstance(key, bytes):
        raise TypeError(...)
    ...
elif isistance(key, str):
    ...
elif isinstance(key, bytes):
    ...
else:
    raise TypeError(...)
```

Reply:
Done. Thanks!